serverless/serverless
# Fix Support cloudformation intrinsic function reference for SNS event

## What did you implement:

Closes #3295.

## How did you implement it:

Added a new property named `topicArn` under the `sns` event type. Together with `topicName` the tuple offers the details necessary to use intrinsic function reference for SNS event.

## How can we verify it:

To verify it, deploy the following JSON:

```
functions:
  ingestFn1:
    handler: handler.ingest
    events:
      - sns:
          topicName: myTopic
          arn:
            Fn::Join:
              - ""
              - - "arn:aws:sns:"
                - Ref: "AWS::Region"
                - ":"
                - Ref: "AWS::AccountId"
                - ":myTopic"

resources:
  Resources:
    myTopic:
      Type: AWS::SNS::Topic
      Properties:
        TopicName: myTopic
        Subscription:
          -
            Endpoint:
              Fn::Join:
                - ""
                - - "arn:aws:sqs:"
                  - Ref: "AWS::Region"
                  - ":"
                  - Ref: "AWS::AccountId"
                  - ":mySqs"
            Protocol: sqs
```

After deploy you'll see the SNS created and linked to the function.

## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
